cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -443,14 +505,26 @@ private void stopProgramIfRunning(Id.Program programId) throws InterruptedExcept
     return namespace;
   }
 
-  private Id.Application validateApplicationId(String namespaceId, String appId)
+  private ApplicationId validateApplicationId(String namespaceId, String appId)
     throws BadRequestException, NamespaceNotFoundException {
+    return validateApplicationVersionId(namespaceId, appId, ApplicationId.DEFAULT_VERSION);
+  }
 
-    Id.Namespace namespace = validateNamespace(namespaceId);
-    try {
-      return Id.Application.from(namespace, appId);
-    } catch (IllegalArgumentException e) {
-      throw new BadRequestException(String.format("Invalid app name '%s': %s", appId, e.getMessage()));
+  private ApplicationId validateApplicationVersionId(String namespaceId, String appId, String versionId)
+    throws BadRequestException, NamespaceNotFoundException {
+    NamespaceId namespace = validateNamespace(namespaceId);
+    if (appId == null) {
+      throw new BadRequestException("Path parameter app-id cannot be empty");
+    }
+    if (!EntityId.isValidId(appId)) {
+      throw new BadRequestException(String.format("Invalid app name '%s'", appId));
+    }
+    if (versionId == null) {
+      throw new BadRequestException("Path parameter version-id cannot be empty");
+    }
+    if (EntityId.isValidId(versionId)) {
```

Comment:
I have the same question in my comments below. I think we should allow versions like `1.0.01`, etc. Let's remove this check.